lemur/plugins/lemur_gcp/endpoints.py

```diff
@@ -100,37 +102,64 @@ def update_target_proxy_cert(project_id, credentials, endpoint, certificate):
     kind = endpoint.type
     if kind not in ("targethttpsproxy", "targetsslproxy") or endpoint.registry_type != "gcp":
         raise NotImplementedError()
-    ssl_client = ssl_certificates.SslCertificatesClient(credentials=credentials)
     # Parses the API name from the certificate body. This is because the certificate's name
     # is different from the name of the certificate that finally gets uploaded to GCP.
     cert_name = certificates.get_name(certificate.body)
-    cert_meta = ssl_client.get(project=project_id, ssl_certificate=cert_name)
+    cert_self_link = certificates.get_self_link(project_id, cert_name)
     if kind == "targethttpsproxy":
         client = target_https_proxies.TargetHttpsProxiesClient(credentials=credentials)
         proxy = client.get(project=project_id, target_https_proxy=endpoint.name)
-        if len(proxy.ssl_certificates) > 1:
-            raise NotImplementedError("GCP endpoints with multiple certificates for SNI are not supported currently.")
-        req = TargetHttpsProxiesSetSslCertificatesRequest()
-        req.ssl_certificates = [cert_meta.self_link]
-        operation = client.set_ssl_certificates(
-            project=project_id,
-            target_https_proxy=endpoint.name,
-            target_https_proxies_set_ssl_certificates_request_resource=req,
-        )
-        operation.result()
+        existing_certs = proxy.ssl_certificates if len(proxy.ssl_certificates) > 0 else []
+        set_target_ssl_proxy_certs(project_id, client, endpoint, [cert_self_link] + existing_certs)
     elif kind == "targetsslproxy":
         client = target_ssl_proxies.TargetSslProxiesClient(credentials=credentials)
         proxy = client.get(project=project_id, target_ssl_proxy=endpoint.name)
-        if len(proxy.ssl_certificates) > 1:
-            raise NotImplementedError("GCP endpoints with multiple certificates for SNI are not supported currently.")
-        req = TargetSslProxiesSetSslCertificatesRequest()
-        req.ssl_certificates = [cert_meta.self_link]
-        operation = client.set_ssl_certificates(
-            project=project_id,
-            target_ssl_proxy=endpoint.name,
-            target_ssl_proxies_set_ssl_certificates_request_resource=req,
-        )
-        operation.result()
+        existing_certs = proxy.ssl_certificates if len(proxy.ssl_certificates) > 0 else []
```

Comment:
A `TargetSslProxy` by definition must always have at least one SSL certificate attached to it so I think this check is unecessary.
```suggestion
        existing_certs = proxy.ssl_certificates
```